include/swift/AST/ASTContext.h

```diff
@@ -863,8 +863,13 @@ class ASTContext {
   /// like `<T>`.
   CanGenericSignature getSingleGenericParameterSignature() const;
   
+  /// Whether our effective Swift version is less than the given version
+  bool isSwiftVersionLessThan(unsigned major, unsigned minor = 0) const {
+    return LangOpts.isSwiftVersionLessThan(major, minor);
+  }
+
   /// Whether our effective Swift version is in the Swift 3 family
-  bool isSwiftVersion3() const { return LangOpts.isSwiftVersion3(); }
+  bool isSwiftVersion3() const { return isSwiftVersionLessThan(4); }
```

Comment:
My original patch removed isSwiftVersion3() entirely, but Ewa wanted something more conservative. I'm open to "< 4" too. What would you prefer?